core/src/main/java/org/apache/calcite/rex/RexCall.java

```diff
@@ -127,7 +143,23 @@ protected final StringBuilder appendOperands(StringBuilder sb) {
           includeType = RexDigestIncludeType.NO_TYPE;
         }
       }
-      sb.append(((RexLiteral) operand).computeDigest(includeType));
+      operandDigests.add(((RexLiteral) operand).computeDigest(includeType));
+    }
+    int totalLength = (operandDigests.size() - 1) * 2; // commas
+    for (String s : operandDigests) {
```

Comment:
How about the operands are empty ?

Reply:
They must not be empty, because `AND()`, `=()` and so on means nothing.